File: hibernate-scala-examples/src/main/scala/org/hibernate/examples/model/PersistentObject.scala
package org.hibernate.examples.model

import java.beans.Transient
import org.hibernate.examples.utils.ToStringHelper

/**
 * kr.debop4s.data.model.PersistentObject
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2013. 12. 24. 오후 2:42
 */
@SerialVersionUID(-7066710546641101707L)
trait PersistentObject extends ValueObject {

    @Transient
    private var persisted: Boolean = false

    def isPersisted = persisted

    protected def setPersisted(v: Boolean) {
        persisted = v
    }

    def onPersist() {
        persisted = true
    }

    def onLoad() {
        persisted = true
    }

    override protected def buildStringHelper: ToStringHelper =
        super.buildStringHelper
            .add("persisted", persisted)
}

File: hibernate-examples/src/main/java/org/hibernate/examples/model/PersistentObject.java
package org.hibernate.examples.model;

/**
 * 영구 저장된 객체인지를 표현하는 Interface입니다.
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2013. 11. 27. 오후 2:37
 */
public interface PersistentObject extends ValueObject {

  boolean isPersisted();

  /**
   * Object 가 저장되었을 때 #isPersisted() 를 true로 갱신한다.
   */
  void onSave();

  /**
   * Object 가 영구저장소에서 로드될 때 #isPersisted() 를 true 로 갱신한다.
   */
  void onLoad();
}
